src/db/result_set.cr

```diff
@@ -55,30 +54,14 @@ module DB
     # Returns the name of the column in `index` 0-based position.
     abstract def column_name(index : Int32) : String
 
-    # Returns the type of the column in `index` 0-based position.
-    # The result is one of `DB::TYPES`.
-    abstract def column_type(index : Int32)
+    # Reads the next column value
+    abstract def read
 
-    def read(t)
-      read?(t).not_nil!
+    # Reads the next column value as a **type**
+    def read(type : Class)
+      type.cast(read)
```

Comment:
@bcardiff I don't know how to easily simplify the specs, I can see `DummyResultSet` receives the query and splits it by commas. I'd leave it like this for now so `read(Class)` is also tested.